helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1861,23 +1861,25 @@ public boolean enableWagedRebalance(String clusterName, List<String> resourceNam
     HelixDataAccessor accessor =
         new ZKHelixDataAccessor(clusterName, new ZkBaseDataAccessor<>(_zkClient));
     PropertyKey.Builder keyBuilder = accessor.keyBuilder();
-    List<IdealState> idealStates = accessor.getChildValues(keyBuilder.idealStates(), true);
+    List<IdealState> idealStates = new ArrayList<>();
+    List<PropertyKey> idealStateKeys = new ArrayList<>();
     List<String> nullIdealStates = new ArrayList<>();
-    for (int i = 0; i < idealStates.size(); i++) {
-      if (idealStates.get(i) == null) {
-        nullIdealStates.add(resourceNames.get(i));
+    for (String resourceName : resourceNames) {
+      PropertyKey key = keyBuilder.idealStates(resourceName);
+      IdealState idealState = accessor.getProperty(key);
```

Comment:
The original batch call is more efficient. Why do we want to change it?
You can still iterate the returned IS list but filter by the resourceNames list. Then check if any item in the resourceNames but not in the returned IS list.

Reply:
I agree, okay.